specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json

```diff
@@ -0,0 +1,846 @@
+        "description": "Gets operation info.",
+        "operationId": "Operations_Get",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "id",
+            "required": true,
+            "description": "The id of the operation",
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9][a-zA-Z0-9._-]{1,62}[a-zA-Z0-9]$",
+            "minLength": 3,
+            "maxLength": 64
+          },
+          {
+            "$ref": "#/parameters/QueryApiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/Operation"
+            }
+          },
+          "default": {
+            "description": "An error occurred.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "type": "string",
+                "description": "Error code"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Operation": {
+            "$ref": "./examples/get_operation.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "BatchSynthesis": {
+      "description": "Bath synthesis task.",
+      "required": [
+        "inputKind",
+        "avatarConfig"
+      ],
+      "type": "object",
+      "properties": {
+        "inputKind": {
+          "$ref": "#/definitions/InputKind"
+        },
+        "inputs": {
+          "description": "Text or SSML inputs, it's allowed up to 1000 items.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BatchSynthesisInput"
+          }
+        },
+        "synthesisConfig": {
+          "$ref": "#/definitions/BatchSynthesisConfig"
+        },
+        "customVoices": {
+          "description": "The custom voice map of voice name and deployment ID .",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9][a-zA-Z0-9._-]{1,62}[a-zA-Z0-9]$",
+            "minLength": 3,
+            "maxLength": 64
+          }
+        },
+        "properties": {
+          "$ref": "#/definitions/BatchSynthesisProperties"
+        },
+        "avatarConfig": {
+          "$ref": "#/definitions/BatchAvatarConfig"
+        },
+        "outputs": {
+          "$ref": "#/definitions/BatchSynthesisOutputs",
+          "readOnly": true
+        },
+        "status": {
+          "$ref": "#/definitions/Status",
+          "readOnly": true
+        },
+        "lastActionDateTime": {
+          "description": "The time-stamp when the current status was entered.",
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "id": {
+          "$ref": "#/definitions/ResourceId"
+        },
```

Comment:
Is this the `{id}` on path? Should it be readOnly in schema?

Reply:
Ah yes it is the id on path. We don't allow user to change it in payload. Updated and marked as readOnly.